### PR TITLE
[ONNX][Autotune] Replace CUDA memory management from CUDART to PyTorch

### PR DESCRIPTION
**Pull request by gcunhase**
### What does this PR do?

**Type of change**: Bug fix

**Overview**: Replace CUDA memory management from CUDART to PyTorch (higher-level API).

### Usage

```python
# Add a code snippet demonstrating how to use this
```

### Testing
1. Added unittests.
2. Tested that this PR does not break https://github.com/NVIDIA/Model-Optimizer/pull/951 or https://github.com/NVIDIA/Model-Optimizer/pull/978

### Before your PR is "*Ready for review*"

Make sure you read and follow [Contributor guidelines](https://github.com/NVIDIA/Model-Optimizer/blob/main/CONTRIBUTING.md) and your commits are signed (`git commit -s -S`).

Make sure you read and follow the [Security Best Practices](https://github.com/NVIDIA/Model-Optimizer/blob/main/SECURITY.md#security-coding-practices-for-contributors) (e.g. avoiding hardcoded `trust_remote_code=True`, using `torch.load(..., weights_only=True)`, avoiding `pickle`, etc.).

- Is this change backward compatible?: ✅
- If you copied code from any other source, did you follow IP policy in [CONTRIBUTING.md](https://github.com/NVIDIA/Model-Optimizer/blob/main/CONTRIBUTING.md#-copying-code-from-other-sources)?: N/A 
- Did you write any new necessary tests?: ✅ 
- Did you update [Changelog](https://github.com/NVIDIA/Model-Optimizer/blob/main/CHANGELOG.rst)?: N/A 

### Additional Information

Summary of changes in `benchmark.py — TensorRTPyBenchmark`:
 | What changed | Before | After |
|---|---|---|
| Imports | `contextlib` + `from cuda.bindings import runtime as cudart` | `import torch` (conditional) |
| Availability flag | `CUDART_AVAILABLE` | `TORCH_CUDA_AVAILABLE = torch.cuda.is_available()` |
| `__init__` guard | checks `CUDART_AVAILABLE or cudart is None` | checks `TORCH_CUDA_AVAILABLE` |
| `_alloc_pinned_host` | `cudaMallocHost` + ctypes address hack, returns `(ptr, arr, err)` | `torch.empty(...).pin_memory()`, returns `(tensor, tensor.numpy())` |
| `_free_buffers` | `cudaFreeHost` + `cudaFree` per buffer | `bufs.clear()` — PyTorch GC handles deallocation |
| `_allocate_buffers` | raw `device_ptr` integers, error-code returns | `torch.empty(..., device="cuda")`, `tensor.data_ptr()` for TRT address |
| `_run_warmup` | `cudaMemcpyAsync` + `cudaStreamSynchronize` | `tensor.copy_(non_blocking=True)` inside `torch.cuda.stream()` |
| `_run_timing` | same cudart pattern | same torch pattern |
| `run` — stream lifecycle | `cudaStreamCreate()` / `cudaStreamDestroy()` | `torch.cuda.Stream()` / `del stream` |
| `run` — stream arg to TRT | raw integer handle | `stream.cuda_stream` (integer property) |
| Error handling | `cudaError_t` return codes | PyTorch raises `RuntimeError`, caught by existing `except Exception` |

Related to https://github.com/NVIDIA/Model-Optimizer/pull/961

## Summary by CodeRabbit

* **Refactor**
  * TensorRT benchmarking migrated from direct CUDA runtime calls to PyTorch CUDA tensors, pinned memory, and CUDA stream primitives — simplifying buffer management, transfers, and timing semantics.
* **Tests**
  * Expanded GPU autotune benchmark tests with broader unit and integration coverage for CUDA/TensorRT paths, pinned-host/device buffering, stream behavior, warmup/timing, and end-to-end latency scenarios.

### CODE REVIEW

File: modelopt/onnx/quantization/autotune/benchmark.py [hunk 1]
```diff
@@ -48,13 +47,13 @@
...
-    except ImportError:
-        with contextlib.suppress(ImportError):
-            from cuda import cudart  # deprecated: prefer cuda.bindings.runtime
+TORCH_CUDA_AVAILABLE = False
+try:
+    import torch
+
+    TORCH_CUDA_AVAILABLE = torch.cuda.is_available()
+except ImportError:
+    pass
```
Comment (kevalmorabia97) [inline modelopt/onnx/quantization/autotune/benchmark.py]: torch is a default dependency and already installed so we dont need this check and can simplify this

File: tests/gpu/onnx/quantization/autotune/test_benchmark.py [hunk 2]
```diff
@@ -13,17 +13,36 @@
...
+import numpy as np
 import pytest
+import torch
 from _test_utils.onnx.quantization.autotune.models import _create_simple_conv_onnx_model
 
+import modelopt.onnx.quantization.autotune.benchmark as bm
 from modelopt.onnx.quantization.autotune import TensorRTPyBenchmark, TrtExecBenchmark
+from modelopt.onnx.quantization.autotune.benchmark import _validate_shape_range
+
+_requires_cuda = pytest.mark.skipif(not bm.TORCH_CUDA_AVAILABLE, reason="CUDA not available")
```
Comment (kevalmorabia97) [inline tests/gpu/onnx/quantization/autotune/test_benchmark.py]: no need for this check. `tests/gpu` always runs on GPU machines

File: modelopt/onnx/quantization/autotune/benchmark.py (hunk 1 above)
Comment (gcunhase) [reply modelopt/onnx/quantization/autotune/benchmark.py]: Done

File: tests/gpu/onnx/quantization/autotune/test_benchmark.py (hunk 2 above)
Comment (gcunhase) [reply tests/gpu/onnx/quantization/autotune/test_benchmark.py]: Done

File: modelopt/onnx/quantization/autotune/benchmark.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (gcunhase) [reply modelopt/onnx/quantization/autotune/benchmark.py]: This code snippet has been removed, outdated.

File: modelopt/onnx/quantization/autotune/benchmark.py [hunk 3]
```diff
@@ -345,9 +344,9 @@ def __init__(
 
         if not TRT_AVAILABLE:
             raise ImportError("TensorRT Python API not available. Please install tensorrt package.")
-        if not CUDART_AVAILABLE or cudart is None:
+        if not TORCH_CUDA_AVAILABLE:
             raise ImportError(
-                "CUDA Runtime (cudart) not available. Please install cuda-python package: pip install cuda-python"
+                "PyTorch with CUDA support not available. Please install torch with CUDA: pip install torch"
             )
```
[the remark this answers is not in the document]
Comment (gcunhase) [reply modelopt/onnx/quantization/autotune/benchmark.py:345]: Done

File: modelopt/onnx/quantization/autotune/benchmark.py [hunk 4]
```diff
@@ -777,16 +719,7 @@ def run(
         finally:
             try:
                 self._free_buffers(inputs + outputs)
```
[the remark this answers is not in the document]
Comment (gcunhase) [reply modelopt/onnx/quantization/autotune/benchmark.py]: Done

File: modelopt/onnx/quantization/autotune/benchmark.py [hunk 5]
```diff
@@ -527,95 +521,57 @@ def _build_engine(
...
     @staticmethod
     def _free_buffers(bufs: list[dict]) -> None:
-        """Free host and device memory for a list of buffer dicts (host_ptr, device_ptr)."""
-        for buf in bufs:
-            if "host_ptr" in buf and buf["host_ptr"] is not None:
-                cudart.cudaFreeHost(buf["host_ptr"])
-            if "device_ptr" in buf and buf["device_ptr"] is not None:
-                cudart.cudaFree(buf["device_ptr"])
+        """Release buffer references; PyTorch handles underlying memory deallocation."""
+        bufs.clear()
```
[the remark this answers is not in the document]
Comment (gcunhase) [reply modelopt/onnx/quantization/autotune/benchmark.py:537]: Fixed
